app/controllers/discussions_messages_controller.rb

```diff
@@ -15,6 +15,10 @@ def destroy
     redirect_back(fallback_location: root_path)
   end
 
+  def preview
+    render json: { preview: Message.new(content: URI.decode(params[:content])).content_html }
```

Comment:
Rails does decode GET query params by default, so I'm guessing it's not liking something in the way we are encoding the param in JS. If we can change that to get rid of this, awesome, if not it's ok too :smile:

Reply:
I removed both this `URI.decode` in Ruby and the `encodeURIComponent()` in JS and everything still works. I was just being extra cautious I guess.

I couldn't break it in any case but we might want to try with all kinds of funky characters just to be sure.